CharlieBackend.Api.UnitTes/ThemeServiceTests.cs

```diff
@@ -0,0 +1,382 @@
+
+            //Assert
+            themeIsUsedExistResult.Error.Code.Should().BeEquivalentTo(ErrorCode.ValidationError);
+        }
+
+        [Fact]
+        public async Task DeleteThemeAsync_NotExistingThemeNotUsed_ShouldReturnNotFound()
+        {
+            //Arrange
+            long notExistingId = 100;
+
+            var themeRepositoryMock = new Mock<IThemeRepository>();
+
+            themeRepositoryMock.Setup(x => x.IsEntityExistAsync(notExistingId))
+                .ReturnsAsync(false);
+
+            _unitOfWorkMock.Setup(x => x.ThemeRepository).Returns(themeRepositoryMock.Object);
+
+            var themeService = new ThemeService(
+                _unitOfWorkMock.Object,
+                _mapper);
+
+            //Act
+            var themeDoesntExistResult = await themeService.DeleteThemeAsync(notExistingId);
+
+            //Assert
+            themeDoesntExistResult.Error.Code.Should().BeEquivalentTo(ErrorCode.NotFound);
+        }
+
+        [Fact]
+        public async Task GetAllThemesAsync_ValidDataPassed_ShouldReturnListOfThemes()
+        {
+            //Arrange
+            var allExistingThemes = new List<Theme>()
+            {
+                new Theme()
+                {
+                    Id = 145,
+                    Name = "Existing_Theme"
+                },
+                new Theme()
+                {
+                    Id = 123,
+                    Name = "Existing_Theme_2"
+                }
+            };
+
+            var expectedResult = new List<ThemeDto>()
+            {
+                new ThemeDto()
+                {
+                    Id = 145,
+                    Name = "Existing_Theme"
+                },
+                new ThemeDto()
+                {
+                    Id = 123,
+                    Name = "Existing_Theme_2"
+                }
+            };
+
+            var themeRepositoryMock = new Mock<IThemeRepository>();
+
+            themeRepositoryMock.Setup(x => x.GetAllAsync())
+                .ReturnsAsync(allExistingThemes);
+
+            _unitOfWorkMock.Setup(x => x.ThemeRepository).Returns(themeRepositoryMock.Object);
+
+            var themeService = new ThemeService(
+                _unitOfWorkMock.Object,
+                _mapper);
+
+            //Act
+            var successResult = await themeService.GetAllThemesAsync();
+
+            //Assert
+            successResult.Data.Should().BeEquivalentTo(expectedResult);
+        }
+
+        [Fact]
+        public async Task GetThemeByNameAsync_ExistingName_ShouldReturnTheme()
+        {
+            //Arrange
+            var existingTheme = new Theme()
+            {
+                Id = 10,
+                Name = "Test_name"
+            };
+
+            var expectedThemeDto = new ThemeDto()
+            {
+                Id = 10,
+                Name = "Test_name"
+            };
+
+            var themeRepositoryMock = new Mock<IThemeRepository>();
```

Comment:
Please, always avoid the repetitive code.
Move this line from each test method into class constructor.